File: QTD.SPA/src/app/_DtoModels/Version_Task_Procedure_Link/Version_Task_Procedure_Link.ts
import { Entity } from '../Entity';
import { Version_Procedure } from '../Version_Procedure/Version_Procedure';
import { Version_Task } from '../Version_Task/Version_Task';

export class Version_Task_Procedure_Link extends Entity {
  version_TaskId!: any;
  version_ProcedureId!: any;
  version_Task!: Version_Task;
  version_Procedure!: Version_Procedure;
}

File: QTD2.Domain/Entities/Core/Version_Task_Procedure_Link.cs
namespace QTD2.Domain.Entities.Core
{
    public class Version_Task_Procedure_Link : Common.Entity
    {
        public int Version_TaskId { get; set; }

        public int Version_ProcedureId { get; set; }

        public string VersionNumber { get; set; }

        public virtual Version_Task Version_Task { get; set; }

        public virtual Version_Procedure Version_Procedure { get; set; }

        public Version_Task_Procedure_Link()
        {
        }

        public Version_Task_Procedure_Link(int version_TaskId, int version_ProcedureId, string versionNumber = "1.0")
        {
            Version_TaskId = version_TaskId;
            Version_ProcedureId = version_ProcedureId;
            VersionNumber = versionNumber;
        }
    }
}
